warp10/src/main/java/io/warp10/script/functions/math/ADDEXACT.java

```diff
@@ -0,0 +1,57 @@
+//
+//   Copyright 2018  Cityzen Data
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions.math;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+
+/**
+ * Check https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html for the documentation of this function.
+ * Last java parameter is on top of the stack.
+ */
+public class ADDEXACT extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public ADDEXACT(String name) {
+    super(name);
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+    Object op0 = stack.pop();
+
+    if (!(op0 instanceof Number)) {
+      throw new WarpScriptException(getName() + " can only operate on numerical values.");
+    }
+
+    Object op1 = stack.pop();
+
+    if (!(op1 instanceof Number)) {
+      throw new WarpScriptException(getName() + " can only operate on numerical values.");
+    }
+
+    if ((op1 instanceof Long || op1 instanceof Integer)
+        && (op0 instanceof Long || op0 instanceof Integer)) {
+      stack.push(Math.addExact(((Number) op1).longValue(), ((Number) op0).longValue()));
```

Comment:
If parameters are both Integer, calling intValue makes more sense

Reply:
The purpose of the *Exact methods is to check for overflow given the type of their parameters, by casting Integer to Long you get rid of the overflow check for Integer.

We could chose to return an int if both parameters were int, this would make it coherent with the other functions and will not be a problem if you never generate Integers on the stack.